lisa/microsoft/testsuites/cpu/cpusuite.py

```diff
@@ -172,150 +163,234 @@ def verify_cpu_offline_network_workload(
-                )
-
-            # verify that the added channels do not handle interrupts on offline cpu
-            lsvmbus_channels = node.tools[Lsvmbus].get_device_channels(force_run=True)
-            for channel in lsvmbus_channels:
-                # verify synthetic network adapter channels align with expected value
-                if channel.class_id == "f8615163-df3e-46c5-913f-f2d2f965ed0e":
-                    log.debug(f"Network synthetic channel: {channel}")
-                    assert_that(channel.channel_vp_map).is_length(
-                        first_current_device_channel
-                    )
+            self._verify_no_irq_on_offline(node, idle, new1)
 
-                # verify that devices do not handle interrupts on offline cpu
-                for channel_vp in channel.channel_vp_map:
-                    assert_that(channel_vp.target_cpu).is_not_in(idle_cpus)
-
-            # reset idle cpu to online
-            set_cpu_state_serial(log, node, idle_cpus, CPUState.ONLINE)
-
-            # reset max and current channel count into original ones
-            # by reloading hv_netvsc driver if hv_netvsc can be reload
-            # otherwise reboot vm
-            if node.tools[KernelConfig].is_built_as_module("CONFIG_HYPERV_NET"):
-                node.tools[Modprobe].reload("hv_netvsc")
-            else:
-                node.tools[Reboot].reboot()
-
-            # change the combined channels count after all cpus online
-            second_channel_count = random.randint(1, min(thread_count, 64))
-            while True:
-                if first_current_device_channel != second_channel_count:
-                    break
-                second_channel_count = random.randint(1, min(thread_count, 64))
-            node.tools[Ethtool].change_device_channels_info(
-                "eth0", second_channel_count
-            )
-            second_current_device_channel = (
-                node.tools[Ethtool].get_device_channels_info("eth0", True)
-            ).current_channels
+            # ---------- Phase 2: CPUs back online ----------
+            set_cpu_state_serial(log, node, idle, CPUState.ONLINE)
+            # Always reboot to ensure network stack is properly reinitialized
+            # after CPU hotplug operations to avoid SSH connection issues
+            node.tools[Reboot].reboot()
+
+            threads2 = node.tools[Lscpu].get_thread_count()
+            dev_max2 = self._read_max_supported(node)
+            upper2 = max(1, min(threads2, dev_max2, 64))
+
+            cur2 = self._read_current(node)
+            if cur2 > upper2:
+                node.tools[Ethtool].change_device_channels_info("eth0", upper2)
+                cur2 = self._read_current(node)
+                log.debug(f"Reduced current channels at phase2: {cur2}")
+
+            tgt2 = self._pick_target_not_eq_current(cur2, upper2)
+            new2 = self._set_channels_with_retry(log, node, tgt2, cur2, upper2)
```

Comment:
Avoid using numeric suffixes; please use meaningful names instead.

Reply:
same comment regarding usage of "tgt", does it mean target?